src/coreclr/src/vm/cominterfacemarshaler.cpp

```diff
@@ -456,6 +385,29 @@ VOID COMInterfaceMarshaler::InitializeExistingComObject(OBJECTREF *pComObj, IUnk
     CreateObjectRef(NeedUniqueObject(), pComObj, ppIncomingIP, /* pIncomingItfMT = */ NULL, /* bIncomingIPAddRefed = */ true);
 }
 
+//--------------------------------------------------------------------------------
+// VOID COMInterfaceMarshaler::InitializeObjectClass()
+//--------------------------------------------------------------------------------
+VOID COMInterfaceMarshaler::InitializeObjectClass(IUnknown *pIncomingIP)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    // If the marshaller's type handle is null, compute what it should be.
+    if (m_typeHandle.IsNull())
+    {
+        // We no longer support the mapping from IUnknown to Type via
+        // IProvideClassinfo in CoreCLR., so if the type handle isn't
+        // set then fallback to the __ComObject type.
+        m_typeHandle = TypeHandle(g_pBaseCOMObject);
```

Comment:
I'm missing what exactly the `IProvideClassInfo` support here meant and why it is fine to remove in this case. Can you elaborate / comment on what scenarios are or are not supported in relation to `IProvideClassInfo`?

Reply:
The [`IProvideClassInfo`](https://docs.microsoft.com/windows/win32/api/ocidl/nn-ocidl-iprovideclassinfo) interface provided an `ITypeInfo` instance that we would then attempt to convert to a `MethodTable` instance. All the logic for mapping and storing `ITypeInfo` <=> `MethodTable` was removed with the `TlbExp` support. I will update the comment. For more context see the removed function `GetClassFromIProvideClassInfo` in this PR.